src/main/data_structure/queue.py

```diff
@@ -0,0 +1,40 @@
+class Queue:
+
+    # Constructor to initiate queue
+    def __init__(self, queue=None):
+        if queue is None:
+            queue = []
+        self.queue = queue
+
+    # push to append an item into queue
+    def push(self, value):
+        self.queue.append(value)
+
+    # front to return starting element from queue
+    def front(self):
+        try:
+            return self.queue[0]
+        except IndexError:
+            raise
+
+    # pop to remove front element from queue
+    def pop(self):
```

Comment:
prev code was okay. we just need to throw an exception like

raise Exception('You are trying to pop an empty Queue')
we don't need try catch here